src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java

```diff
@@ -46,6 +46,32 @@ public LinuxOperatingSystem() {
         this.version = (osRelease != null) ? new LinuxOSVersionInfoEx(osRelease) : new LinuxOSVersionInfoEx();
     }
 
+    @Override
+    public long getOpenDescriptors() {
+        return getDescriptors(0);
+    }
+
+    @Override
+    public long getMaxDescriptors() {
+        return getDescriptors(2);
+    }
+
+    private long getDescriptors(int index) {
+        if (new File("/proc/sys/fs/file-nr").exists()) {
+            try {
+                List<String> osDescriptors = FileUtil.readFile("/proc/sys/fs/file-nr");
+                for (String line : osDescriptors) {
+                    line = line.replaceAll("[^0-9]+", " ");
+                    String [] splittedLine = line.split(" ");
+                    return Long.parseLong(splittedLine[index]);

```

Comment:
Need to do argument check for index between 0 and 2 before this point, where an array index out of bounds will be much more confusing to debug.